### PR TITLE
refactor: helm chart release automation

### PR DESCRIPTION
**Pull request by royhadad**

### CODE REVIEW

File: .github/workflows/helm-docs-validation.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        run: |
+          curl -LO https://github.com/norwoodj/helm-docs/releases/download/v1.11.0/helm-docs_1.11.0_Linux_x86_64.tar.gz
+          tar -xvf helm-docs_1.11.0_Linux_x86_64.tar.gz
+          sudo mv helm-docs /usr/local/bin/
+
+      - name: Generate Helm Docs
+        run: make generate-helm-docs
+
+      - name: Check for file changes
+        run: |
```
Comment (TzlilSwimmer123) [inline .github/workflows/helm-docs-validation.yaml:24]: maybe it's good to add `shell: bash`

File: Makefile [hunk 2]
```diff
@@ -81,3 +81,16 @@ helm-template-staging:
...
 	--debug && \
 	make change-ping-uninstall-url-to-production
+
+# in order to run the command, first install helm-docs by running: "brew install norwoodj/tap/helm-docs"
+# https://github.com/norwoodj/helm-docs
+generate-helm-docs:
+	helm-docs \
+	--sort-values-order=file \
+	--output-file ./README.md \
+	--template-files=./charts/datree-admission-webhook/README.md.gotmpl \
```
Comment (TzlilSwimmer123) [inline Makefile:91]: why `.gotmpl` is needed?

File: README.md.gotmpl [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Datree Admission Webhook
```
Comment (TzlilSwimmer123) [inline README.md.gotmpl:1]: why this file was added?

File: charts/datree-admission-webhook/values.yaml [hunk 4]
```diff
@@ -71,38 +73,46 @@ securityContext:
...
 resources: { }
 # limits:
 #   cpu: 200m
 #   memory: 256Mi
 # requests:
 #   cpu: 200m
 #   memory: 48Mi
+# -- Used to select on which node a pod is scheduled to run
 nodeSelector: { }
+# -- Offers more expressive syntax for fine-grained control of how Pods are scheduled to specific nodes
```
Comment (TzlilSwimmer123) [inline charts/datree-admission-webhook/values.yaml:86]: comments feels redundant because it's something that k8s offers

**Review by TzlilSwimmer123 — COMMENTED** [review]
overall looks good, left some questions

File: .github/workflows/helm-docs-validation.yaml (hunk 1 above)
Comment (royhadad) [reply .github/workflows/helm-docs-validation.yaml:24]: It's not critical production code, and it gets the job done, so I think it's fine.
what does `shell: bash` do?

File: Makefile (hunk 2 above)
Comment (royhadad) [reply Makefile:91]: It's a file format :)

https://github.com/norwoodj/helm-docs/tree/master#:~:text=The%20markdown%20generation,reasonably%20formatted%20README.

File: README.md.gotmpl (hunk 3 above)
Comment (royhadad) [reply README.md.gotmpl:1]: This is the file that tells the "helm docs" library how to render the README.md file

**Review by TzlilSwimmer123 — APPROVED** [review]
🤓

**Pull request merged**